pkg/kubecost/asset.go

```diff
@@ -3211,24 +3211,27 @@ func (as *AssetSet) ReconciliationMatchMap() map[string]map[string]Asset {
 			continue
 		}
 
-		if _, ok := matchMap[props.ProviderID]; !ok {
-			matchMap[props.ProviderID] = make(map[string]Asset)
+		// we can't guarantee case in providerID for Azure provider to have map working for all providers,
```

Comment:
Why isn't casing guaranteed, does azure lack case-sensitivity?

Reply:
Yes, at this time we cannot guarantee casing in providerID. It can be Pascal case or all lower case or combination of both in the ProviderID for Azure resources.